packages/superset-ui-encodable/src/fillers/completeChannelDef.ts

```diff
@@ -1,27 +1,51 @@
-import { ChannelDef } from '../types/ChannelDef';
+import { ChannelDef, NonValueDef } from '../types/ChannelDef';
 import { ChannelType } from '../types/Channel';
-import { isFieldDef } from '../typeGuards/ChannelDef';
+import { isFieldDef, isValueDef, isTypedFieldDef } from '../typeGuards/ChannelDef';
 import completeAxisConfig, { CompleteAxisConfig } from './completeAxisConfig';
 import completeScaleConfig, { CompleteScaleConfig } from './completeScaleConfig';
-import { Value } from '../types/VegaLite';
+import { Value, ValueDef, Type } from '../types/VegaLite';
+import inferFieldType from './inferFieldType';
 
-type CompleteChannelDef<Output extends Value = Value> = Omit<
-  ChannelDef,
+export interface CompleteValueDef<Output extends Value = Value> extends ValueDef<Output> {
```

Comment:
is this a `type` or a `variable`? I don't quite understand it, just wanted to check it was intentional.

Reply:
It is a `type`.